modules/core/src/utils/deep-equal.ts

```diff
@@ -1,17 +1,17 @@
 // Partial deep equal (only recursive on arrays)
-export function deepEqual(a: any, b: any): boolean {
+export function deepEqual(a: any, b: any, recursive: boolean = false): boolean {
   if (a === b) {
     return true;
   }
   if (!a || !b) {
     return false;
   }
-  for (const key in a) {
+  const keys = [...new Set([...Object.keys(a), ...Object.keys(b)])];
```

Comment:
Given the perfomance profile of deck.gl (i.e. we do "insane" amounts of comparisons to minimize the number of actual updates), generally we want comparison to be as fast as we can make it. 

Something we particularly want to avoid during comparison is creation of a lot of small objects that need to be allocated and then garbage collected.

While this code may look easier to read (something I normally prioritize), we should revert.

Adding benchmarks for our comparison functions wouldn't be unreasonable.

Reply:
I've added a bench test and compared the performance against the old version. All the results stay broadly the same (except the recursive comparison as expected)

```
Same object               65.5M/s -> 66.7M/s
Different object          10.4M/s -> 10.7M/s
Same array                51.8M/s -> 55.3M/s
Different array           3.76M/s -> 3.71M/s
Nested object             12.7M/s -> 11.5M/s
Nested object recursive   12.7M/s -> 4.48M/s
```